ios/Classes/ChirpFlutterPlugin.m

```diff
@@ -0,0 +1,366 @@
+     * sendingBlock is called when a send event begins.
+     * The data argument contains the payload being sent.
+     *----------------------------------------------------------------------------*/
+    [weakSelf.sendingStreamHandler send:[FlutterStandardTypedData typedDataWithBytes:data]
+                                channel:[NSNumber numberWithInteger:channel]];
+  }];
+
+  [weakSelf.chirp setSentBlock:^(NSData * _Nonnull data, NSUInteger channel)
+  {
+    /*------------------------------------------------------------------------------
+     * sentBlock is called when a send event has completed.
+     * The data argument contains the payload that was sent.
+     *----------------------------------------------------------------------------*/
+    [weakSelf.sentStreamHandler send:[FlutterStandardTypedData typedDataWithBytes:data]
+                             channel:[NSNumber numberWithInteger:channel]];
+  }];
+
+  [weakSelf.chirp setReceivingBlock:^(NSUInteger channel)
+  {
+    /*------------------------------------------------------------------------------
+     * receivingBlock is called when a receive event begins.
+     * No data has yet been received.
+     *----------------------------------------------------------------------------*/
+    [weakSelf.receivingStreamHandler send:[NSNumber numberWithInteger:channel]];
+  }];
+
+  [weakSelf.chirp setReceivedBlock:^(NSData * _Nullable data, NSUInteger channel)
+   {
+     /*------------------------------------------------------------------------------
+      * receivedBlock is called when a receive event has completed.
+      * If the payload was decoded successfully, it is passed in data.
+      * Otherwise, data is null.
+      *----------------------------------------------------------------------------*/
+    [weakSelf.receivedStreamHandler send:[FlutterStandardTypedData typedDataWithBytes:data]
+                                 channel:[NSNumber numberWithInteger:channel]];
+   }];
+}
+
+- (void)init:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString *key = call.arguments[@"key"];
+  NSString *secret = call.arguments[@"secret"];
+  self.chirp = [[ChirpSDK alloc] initWithAppKey:key
+                                      andSecret:secret];
+  if (self.chirp) {
+    result(nil);
+  } else {
+    result([FlutterError errorWithCode:@"-1"
+                               message:@"Failed to initialise ChirpSDK"
+                               details:nil]);
+  }
+}
+
+- (void)version:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  result([self.chirp version]);
+}
+
+- (void)setConfig:(FlutterMethodCall*)call result:(FlutterResult)result {
+  if (![self isInitialised:call result:result]) return;
+  NSString *config = call.arguments;
+  NSError *error = [self.chirp setConfig:config];
+  if (error) {
+    result([FlutterError errorWithCode:[NSString stringWithFormat:@"%ld", (long)error.code]
```

Comment:
Can we use `handleError` here?

Reply:
I was thinking that calling `result` in `handleError` would mean that `setCallbacks` would never get called, but I suppose the function will always finish, result is just sending the message back through the method channel.

Reply:
Sure, you still need to keep the if/else, but just for consistency it would be great to use `handleError` everywhere so when you'll need to modify error handler you'll have to do it in one place.